lib/march_hare/channel.rb

```diff
@@ -614,10 +614,14 @@ def basic_get(queue, auto_ack)
       end
     end
 
-    def basic_consume(queue, auto_ack, consumer)
```

Comment:
Wow, in a decade with Ruby I haven't seen a method where arguments with default values weren't at the end of the argument list. That indeed works as you describe:

```
2.2.2 :001 > def abc(a, b, c = nil, d)
2.2.2 :002?>   [a, b, c, d]
2.2.2 :003?>   end
 => :abc
2.2.2 :004 > abc(1, 2, 3, 4)
 => [1, 2, 3, 4]
2.2.2 :005 > abc(1, 2, 4)
 => [1, 2, nil, 4]
```

Perfect, I'm going to QA it right away then. Thank you.